lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -499,6 +500,12 @@ MemoryBehaviorVisitor::visitDestroyValueInst(DestroyValueInst *DVI) {
   return MemBehavior::MayHaveSideEffects;
 }
 
+MemBehavior MemoryBehaviorVisitor::visitEndBorrowInst(EndBorrowInst *EBI) {
+  if (!EA->canEscapeTo(V, EBI))
+    return MemBehavior::None;
+  return MemBehavior::MayRead;
+}
+
```

Comment:
Can you also add MemBehavior for load_borrow / begin_borrow with tests. And return MemBehavior::None for end_borrow of a `begin_borrow` ? 

These changes and tests are in https://github.com/apple/swift/pull/35866. I did not merge that PR because it opens up new issues in TempRVO which need to be addressed as well.